lisa/sut_orchestrator/azure/platform_.py

```diff
@@ -1794,14 +1817,40 @@ def _resource_sku_to_capability(  # noqa: C901
         else:
             node_space.disk.disk_controller_type.add(schema.DiskControllerType.SCSI)
 
+        # If EphemeralOSDisk is supported, then check for the placement type
         if azure_raw_capabilities.get("EphemeralOSDiskSupported", None) == "True":
-            # Check if CachedDiskBytes is greater than 30GB
-            # We use diff disk as cache disk for ephemeral OS disk
+            node_space.disk.os_disk_type.add(schema.DiskType.Ephemeral)
+            # Add the EphemeralDiskPlacementType
+            ephemeral_disk_placement_types = azure_raw_capabilities.get(
+                "SupportedEphemeralOSDiskPlacements", None)
+            if ephemeral_disk_placement_types:
+                for allowed_type in ephemeral_disk_placement_types.split(","):
+                    try:
+                        node_space.disk.ephemeral_disk_placement_type.add(
+                            schema.EphemeralDiskPlacementType(allowed_type)
+                        )
+                    except ValueError:
+                        self._log.error(
+                            f"'{allowed_type}' is not a known Ephemeral Disk Placement"
+                            f" Type "
+                            f"({[x for x in schema.EphemeralDiskPlacementType]})"
+                        )
+
+            # EphemeralDiskPlacementType can be - ResourceDisk, CacheDisk or NvmeDisk.
+            # Depending on that, "CachedDiskBytes" may or may not be found in
+            # capabilities.
+            # refer
+            # https://learn.microsoft.com/en-us/azure/virtual-machines/ephemeral-os-disks-faq
+            resource_disk_bytes = azure_raw_capabilities.get("MaxResourceVolumeMB", 0)
             cached_disk_bytes = azure_raw_capabilities.get("CachedDiskBytes", 0)
-            cached_disk_bytes_gb = int(int(cached_disk_bytes) / 1024 / 1024 / 1024)
-            if cached_disk_bytes_gb >= 30:
-                node_space.disk.os_disk_type.add(schema.DiskType.Ephemeral)
-                node_space.disk.os_disk_size = cached_disk_bytes_gb
+            nvme_disk_bytes = azure_raw_capabilities.get("NvmeDiskSizeInMiB", 0)
+            if nvme_disk_bytes:
```

Comment:
Does it mean Nvme disk size is higher than normal disk size? Is it possible the two settings co-exists? If they cannot be co-exists, it needs to check and assert.

Reply:
> Does it mean Nvme disk size is higher than normal disk size? Is it possible the two settings co-exists? If they cannot be co-exists, it needs to check and assert.

NVMe disk size is higher and preferred if it is present. I know that "CacheDisk" and "ResourceDisk" can co-exist. As of now, I haven't seen "NVMeDisk" co-exist with any other, but I am not ruling it out

>>>>>>
(Get-AzComputeResourceSku -Location EastUS2euap | ? {$_.Name -like "Standard_D2s_v3"}).Capabilities

Name                                                                        Value
----                                                                             -----
MaxResourceVolumeMB                                          16384
OSVhdSizeMB                                                          1047552
vCPUs                                                                       2
MemoryPreservingMaintenanceSupported            True
HyperVGenerations                                                 V1,V2
SupportedEphemeralOSDiskPlacements                ResourceDisk,CacheDisk
CachedDiskBytes                                                     53687091200